compiler/noirc_frontend/src/monomorphization/printer.rs

```diff
@@ -455,6 +481,25 @@ impl AstPrinter {
         Ok(())
     }
 
+    /// Instead of printing a call to the print oracle as a regular function,
+    /// print it in a way that makes it look like Noir: without the type
+    /// information and bool flags.
+    fn print_println(&mut self, args: &[Expression], f: &mut Formatter) -> std::fmt::Result {
+        assert_eq!(args.len(), 4, "print has 4 arguments");
+        let Expression::Literal(Literal::Bool(with_newline)) = args[0] else {
+            unreachable!("the first arg of print is a bool");
+        };
+        if with_newline {
+            write!(f, "println")?;
+        } else {
+            write!(f, "print")?;
+        }
+        write!(f, "(")?;
+        self.print_expr(&args[1], f)?;
```

Comment:
Would it be useful to include the other two arguments when printing this?

```
        // The print oracle takes 2 parameters: the newline marker and the value,
        // but it takes 2 more arguments: the type descriptor and the format string marker.
```

Reply:
I'll extend the comment to explain that these parameters are automatically added by the monomorphizer, but they are not parameters of `println` itself.